test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go

```diff
@@ -225,20 +336,101 @@ func (r *InMemoryMachineReconciler) reconcileNormal(ctx context.Context, cluster
+		}
+
+		// NOTE: for the first control plane machine we might create the node before etcd and API server pod are running
+		// but this is not an issue, because it won't be visible to CAPI until the API server start serving requests.
+		if err := cloudClient.Create(ctx, node); err != nil && !apierrors.IsAlreadyExists(err) {
+			return ctrl.Result{}, errors.Wrapf(err, "failed to create Node")
+		}
 	}
 
-	// NOTE: this can probably go up, after the VM is created
 	inMemoryMachine.Spec.ProviderID = &node.Spec.ProviderID
 	inMemoryMachine.Status.Ready = true
 
+	conditions.MarkTrue(inMemoryMachine, infrav1.NodeProvisionedCondition)
+	return ctrl.Result{}, nil
+}
+
+func (r *InMemoryMachineReconciler) reconcileNormalETCD(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, inMemoryMachine *infrav1.InMemoryMachine) (ctrl.Result, error) {
+	// No-op if the machine is not a control plane machine.
 	if !util.IsControlPlaneMachine(machine) {
 		return ctrl.Result{}, nil
 	}
 
-	// If there is not yet an etcd member listener for this machine.
+	// No-op if the Node is not provisioned yet
+	if !conditions.IsTrue(inMemoryMachine, infrav1.NodeProvisionedCondition) {
+		return ctrl.Result{}, nil
+	}
+
+	// Wait for the etcd pod to start up; etcd pod start happens a configurable time after the Node is provisioned.
+	provisioningDuration := time.Duration(0)
+	if inMemoryMachine.Spec.Behaviour != nil && inMemoryMachine.Spec.Behaviour.Etcd != nil {
+		x := inMemoryMachine.Spec.Behaviour.Etcd.Provisioning
+
+		provisioningDuration = x.StartupDuration.Duration
+		jitter, _ := strconv.ParseFloat(x.StartupJitter, 64)
+		if jitter > 0.0 {
+			provisioningDuration += time.Duration(rand.Float64() * jitter * float64(provisioningDuration)) //nolint:gosec // Intentionally using a weak random number generator here.
+		}
+	}
+
+	start := conditions.Get(inMemoryMachine, infrav1.NodeProvisionedCondition).LastTransitionTime
+	now := time.Now()
+	if now.Before(start.Add(provisioningDuration)) {
+		conditions.MarkFalse(inMemoryMachine, infrav1.EtcdProvisionedCondition, infrav1.EtcdProvisioningReason, clusterv1.ConditionSeverityInfo, "")
+		return ctrl.Result{RequeueAfter: start.Add(provisioningDuration).Sub(now)}, nil
+	}
+
+	// Compute the resource group unique name.
+	// NOTE: We are using reconcilerGroup also as a name for the listener for sake of simplicity.
+	resourceGroup := klog.KObj(cluster).String()
+	cloudClient := r.CloudManager.GetResourceGroup(resourceGroup).GetClient()
+
+	// Create the etcd pod
+	// TODO: consider if to handle an additional setting adding a delay in between create pod and pod ready
 	etcdMember := fmt.Sprintf("etcd-%s", inMemoryMachine.Name)
+	etcdPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceSystem,
+			Name:      etcdMember,
+			Labels: map[string]string{
+				"component": "etcd",
+				"tier":      "control-plane",
+			},
```

Comment:
Did we intentionally drop the TODOs?
```go
			// TODO: read this from existing etcd pods, if any.
			"etcd.inmemory.infrastructure.cluster.x-k8s.io/cluster-id": fmt.Sprintf("%d", rand.Uint32()), //nolint:gosec // weak random number generator is good enough here
			"etcd.inmemory.infrastructure.cluster.x-k8s.io/member-id":  fmt.Sprintf("%d", rand.Uint32()), //nolint:gosec // weak random number generator is good enough here
			// TODO: set this only if there are no other leaders.
```

Reply:
Not sure I understand this. The TODOs where about that additional replicas should use the same cluster-id and something about the leader thing.

Reply:
@sbueringer fixed, rebased, squashed